R/zchunk_LA154.Transport.R

```diff
@@ -32,71 +31,215 @@ module_gcam.usa_LA154.Transport_DISABLED <- function(command, ...) {
+                          "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM",
+                          "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA",
+                          "WV", "WI", "WY")
+
+        L154.out_mpkm_R_trn_nonmotor_Yh %>%
+          rename(value_mode = value) %>%
+          filter(GCAM_region_ID == gcam.USA_CODE) %>%
+          spread(year, value_mode) %>%
+          select(-GCAM_region_ID, -mode) ->
+          nation_data
+
+        Pop_state_share %>%
+          filter(year %in% HISTORICAL_YEARS) %>%
+          mutate(mode = "Walk") %>%
+          spread(year, value_share) %>%
+          mutate(state = factor(state, levels = old_df_order)) %>% # Need to reorder based on old dataframe
+          arrange(state) ->
+          State_walk
+
+        Pop_state_share %>%
+          filter(year %in% HISTORICAL_YEARS) %>%
+          mutate(mode = "Cycle") %>%
+          spread(year, value_share) %>%
+          mutate(state = factor(state, levels = old_df_order)) %>% # Need to reorder based on old dataframe
+          arrange(state) %>%
+          bind_rows(State_walk) %>% # Need to reorder based on old dataframe
+          select(-state, -mode) ->
+          state_share_data
+
+        # This is the underlying code of the function, apportion_to_states
+        data_final <- state_share_data[  ] *
+          nation_data[ rep( 1:nrow( nation_data ), length.out = nrow( state_share_data ) ),  ]
+
+        data_final <- as_tibble(data_final) # Need to convert the dataframe back to a tibble
+
+        data_final %>%
+          filter(row_number() <= 51) %>%
+          mutate(mode = "Cycle") %>%
+          mutate(state = old_df_order) ->
+          State_cycle
+
+        data_final %>%
+          filter(row_number() > 51) %>%
+          mutate(mode = "Walk") %>%
+          mutate(state = old_df_order) %>%
+          bind_rows(State_cycle) %>%
+          gather(year, value, -state, -mode) %>%
+          select(state, mode, year, value) %>%
+          mutate(year = as.integer(year)) ->
+          L154.out_mpkm_state_trn_nonmotor_Yh
+
+        } else {
+
+        # Now we can use these shares to allocate the national data across the states
+        L154.out_mpkm_R_trn_nonmotor_Yh %>%
+          rename(value_mode = value) %>%
+          filter(GCAM_region_ID == gcam.USA_CODE) %>%
+          # Number of rows will change by adding states, so left_join_error_no_match cannot be used
+          left_join(Pop_state_share, by = "year") %>%
+          mutate(value = value_mode * value_share) %>% # Apportioning across the modes using the share data
+          filter(year %in% HISTORICAL_YEARS) %>% # Ensuring within historical period
+          select(state, mode, year, value) %>%
+          mutate(year = as.integer(year)) ->
+          L154.out_mpkm_state_trn_nonmotor_Yh
```

Comment:
The sum is the same, as just the allocation was applied wrongly. For every other state, the values for the two modes of transportation are switched. The remainder states were correct.